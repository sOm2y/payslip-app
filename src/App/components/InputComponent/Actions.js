//@flow

import React from 'react';
import Button from 'material-ui/Button';

class Actions extends React.PureComponent<{ onClickOk: Function, onClickReset: Function }> {
  render() {
    const { onClickOk, onClickReset } = this.props;

    return (
      <div
        style={{
          display: 'flex',
          justifyContent: 'space-between'
        }}
      >
        <Button className="btn_reset" onClick={onClickReset}>
          Reset
        </Button>

        <Button className="btn_ok" color="primary" raised onClick={onClickOk}>
          Ok
        </Button>
      </div>
    );
  }
}

export default Actions;
