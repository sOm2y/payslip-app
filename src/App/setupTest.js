//@flow

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// enable using of expect(wrapper).toContainReact(<SomeComponent/>)
import 'jest-enzyme';

configure({ adapter: new Adapter() });
