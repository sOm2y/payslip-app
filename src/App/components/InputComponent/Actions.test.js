//@flow

import React from 'react';
import Actions from './Actions';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import '../../setupTest'; // enzyme setup

describe('Actions', () => {
  it('should renders without crashing even for shallow renderer', () => {
    shallow(<Actions onClickOk={() => {}} onClickReset={() => {}} />);
  });

  it('should have ok/reset action', () => {
    const ok = jest.fn();
    const reset = jest.fn();
    const actions = shallow(<Actions onClickOk={ok} onClickReset={reset} />);
    actions.find('.btn_ok').simulate('click');
    actions.find('.btn_reset').simulate('click');

    expect(ok).toHaveBeenCalledTimes(1);
    expect(reset).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<Actions onClickOk={() => {}} onClickReset={() => {}} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
