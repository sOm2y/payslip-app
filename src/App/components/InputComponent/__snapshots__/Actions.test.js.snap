// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions should match snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
    }
  }
>
  <button
    className="MuiButtonBase-root-15 MuiButton-root-1 btn_reset"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label-3"
    >
      Reset
    </span>
    <span
      className="MuiTouchRipple-root-17"
    />
  </button>
  <button
    className="MuiButtonBase-root-15 MuiButton-root-1 MuiButton-raised-8 MuiButton-raisedPrimary-10 btn_ok"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label-3"
    >
      Ok
    </span>
    <span
      className="MuiTouchRipple-root-17"
    />
  </button>
</div>
`;
